launch/launch/actions/execute_local.py

```diff
@@ -714,6 +750,18 @@ def execute(self, context: LaunchContext) -> Optional[List[LaunchDescriptionEnti
             else:
                 self.__stdout_logger, self.__stderr_logger = \
                     launch.logging.get_output_loggers(name, self.__output)
+            # Update the cmd to respect split_arguments option.
+            if evaluate_condition_expression(context, self.__split_arguments):
+                if self.__shell:
+                    self.__logger.debug("Ignoring 'split_arguments=True' because 'shell=True'.")
+                else:
+                    self.__logger.debug("Splitting arguments because 'split_arguments=True'.")
+                    expanded_cmd = []
+                    assert self.__process_event_args is not None
+                    for token in self.__process_event_args['cmd']:
+                        expanded_cmd.extend(shlex.split(token, posix=g_is_windows))
```

Comment:
One thing I'm confused by is...

It seems that the posix argument is default `True` according to the [`shlex` docs](https://docs.python.org/3/library/shlex.html#shlex.split)

I'm not sure why specifying this changes the behavior for windows? :o
If I read this correctly, it'll only change the behavior for non-windows platforms (setting posix to false for those platforms.)

Reply:
Well, I didn't test on Windows, so it's probably wrong? ... Let me add a test for the opposite for Posix. My guess is that it will still fail on Windows and doesn't matter what it is on Linux.

It's a good catch, sorry I'll have another look.

Reply:
Speaking of which, it might also be good to add a testcase with a backslash in your tests :thinking:

Reply:
You might be able to get by with just negating the posix argument:
```
>>> shlex.split("a\\w")
['aw']
>>> shlex.split("a\\w", posix=False)
['a\\w']
>>> shlex.split("a\\w", posix=False)
['a\\w']
```

Reply:
```suggestion
                        expanded_cmd.extend(shlex.split(token, posix=(not g_is_windows)))
```

Reply:
+1

Reply:
Ok, [0c5a017](https://github.com/ros2/launch/pull/711/commits/0c5a017e817122b01ea1bcd82d57c5e8ba94feee) should address this.